### DISCUSSION TITLE
Breaking changes in Rsbuild v0.6.0

### DISCUSSION THREAD

**Discussion by chenjiahan** [web-infra-dev/rsbuild#1970] [RFC Discussions]
Rsbuild v0.6 will be released alongside Rspack v0.6. It will retain the same minor version number as Rspack.

**Comment by chenjiahan** [comment]
## Upgrade Rspack v0.6.0

Rsbuild v0.6 has upgraded the dependent`@rspack/core` to v0.6.0.

Please note that Rspack v0.6.0 contains some breaking changes, see https://github.com/web-infra-dev/rspack/releases/tag/v0.6.0 for more details.

**Reply by chenjiahan** [reply]
https://github.com/web-infra-dev/rsbuild/pull/1985

**Comment by chenjiahan** [comment]
## Default server port to 3000

The default value of `server.port` is switched from `8080` to `3000`.

3000 is is commonly used for web development purposes and is used by create-react-app. Switching the default port can help users avoid port conflicts caused by 8080.

**Reply by chenjiahan** [reply]
ref: https://teenycoders.com/what-is-localhost-explaining-localhost-3000-and-localhost-8080/

**Reply by chenjiahan** [reply]
https://github.com/web-infra-dev/rsbuild/pull/2043

**Reply by xc2** [reply]
there's also a known and wired issue to do with ws on 8080/tcp: #1541

might be port reuse thing.

**Comment by chenjiahan** [comment]
## Enable error overlay by default

Starting from v0.6, the default value of `dev.client.overlay` is `true`. This means that the error overlay is displayed by default when there are compilation errors.

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/bef558ba-5ce2-415d-a5ca-e1730f427b5c)

You can disable it by setting `dev.client.overlay` to `false`:

```js
// rsbuild.config.ts
export default defineConfig({
  dev: {
    client: {
      overlay: false,
    },
  },
});
```

**Reply by chenjiahan** [reply]
https://github.com/web-infra-dev/rsbuild/pull/2050
